### PR TITLE
Generate `connectors.md`

### PR DESCRIPTION
**Pull request by evantahler**
This PR adds an automatically-generated `connectors.md` file at the root of this directory.  It takes the information contained within the actor_definition files and builds a human-readable markdown table any time a connector is added or updated.

This answers the question - "What's the official list of connectors"?

In the future, once all connector metadata lives within this repo, we can add cloud status and additional information.   This happens automatically as part of the `/publish` workflow, keyed as a dependency of `:airbyte-config:init:processResources`

**[View the file here](https://github.com/airbytehq/airbyte/blob/evan/connectors.md/connectors.md)**

Screenshot 2023-02-22 at 3 44 48 PM

### CODE REVIEW

**Review by bnchrch — APPROVED** [review]
Love the spirit and presentation of this. Let merge it in.

For context, in the future this file may take on a different "form" as metadata likely willl:
1. Make the catalog the primary reference point and the definition yamls will go away.
2. Move the generation of the catalog outside of CI/CD and not checked into the repository.

**Review by pedroslopez — APPROVED** [review]
_this is absolutely beautiful_

File: airbyte-config/specs/src/main/java/io/airbyte/config/specs/ConnectorMarkdownGenerator.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        final Path seedRoot = Path.of(parsed.getOptionValue(SEED_ROOT_OPTION.getOpt()));
+        final String outputFileName = parsed.getOptionValue(OUTPUT_FILENAME_OPTION.getOpt());
+
+        final ConnectorMarkdownGenerator mdGenerator = new ConnectorMarkdownGenerator();
+        mdGenerator.run(outputRoot, seedRoot, outputFileName);
+    }
+
+    public void run(final Path outputRoot, final Path seedRoot, final String outputFileName) {
+        final List<JsonNode> destinationDefinitionsJson = getSeedJson(seedRoot, SeedConnectorType.DESTINATION.getDefinitionFileName());
+        final List<JsonNode> sourceDefinitionsJson = getSeedJson(seedRoot, SeedConnectorType.SOURCE.getDefinitionFileName());
```
[the remark this answers is not in the document]
Comment (evantahler) [reply airbyte-config/specs/src/main/java/io/airbyte/config/specs/ConnectorMarkdownGenerator.java:45]: I very much hope this file one day is built by our metadata service.  I view this as a transitionary step - if we can get folks looking here, and not the actor definitions, than the pain of removing them is lessened.

File: airbyte-config/specs/src/main/java/io/airbyte/config/specs/ConnectorMarkdownGenerator.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        bodyParts.add("");
+        bodyParts.add("## Destinations");
+        bodyParts.add("");
+        bodyParts.add(buildMarkdownTable(destinationDefinitionsJson, "Destination"));
+
+        return String.join("\r\n", bodyParts);
+    }
+
+    @SuppressWarnings("PMD")
+    private String buildMarkdownTable (List<JsonNode> definitions, String type) {
```
[the remark this answers is not in the document]
Comment (evantahler) [reply airbyte-config/specs/src/main/java/io/airbyte/config/specs/ConnectorMarkdownGenerator.java:79]: Done! 732307bd468cdfe8f5247a39e579910d18b5f512

**Comment by subodh1810** [comment]
@evantahler the file is breaking CI. It fails the check `Ensure no file change from regenerating connector definitions/specs` in the connector base build https://github.com/airbytehq/airbyte/actions/runs/4253047335/jobs/7397428169

The timestamp for `__Generated` will never be the same. We need to ignore the file 
cc @sergio-ropero

**Comment by evantahler** [comment]
This PR will catch the problem above in tests for the future https://github.com/airbytehq/airbyte/pull/23370

**Comment by evantahler** [comment]
https://github.com/airbytehq/airbyte/pull/23372, https://github.com/airbytehq/airbyte/pull/23369, https://github.com/airbytehq/airbyte/pull/23393, and https://github.com/airbytehq/airbyte/pull/23392 are all fixes for this PR

**Pull request merged**